include/mxnet/gpu_swap_history.h

```diff
@@ -41,35 +40,38 @@ class MemHistory {
     size_t record_step;
     size_t size;
   };
+  struct DeviceHistory {
+    std::map<handle_id_t, std::vector<MemRecord>> handle_history;
+    std::vector<MemRecord> ordered_history;
+    // FIXME(fegin): Do we really need seperate variables to do LRU?
```

Comment:
I think history list + pointer map is the most efficient way of implementing LRU. Maybe we can implement LRU using just  ordered_history but then we will need to do lots of searching which is slower

Reply:
I agree. If we only want to use LRU when history is not collected, a naive LRU is enough. But if we want to compare LRU with other algorithms in later loops, an efficient LRU is more helpful.

Reply:
I'll remove the fixme